src/transaction.js

```diff
@@ -270,6 +272,91 @@ export default class Transaction {
         this.acc = new Account(network_byte);
     }
 
+    tokenContractDataGen(amount, unity, des) {
+        let max = BigNumber(amount).multipliedBy(BigNumber(unity)).toString()
+        let data = [
+            { type: Constants.AMOUNT_TYPE, value: max },
+            { type: Constants.AMOUNT_TYPE, value: BigNumber(unity).toString() },
+            { type: Constants.SHORTTEXT_TYPE, value: des}
+            ]
+        return data
+    }
+
+    paymentContractDataGen(token_id) {
+        return [{ type: Constants.TOKEN_ID_TYPE, value: token_id}]
+    }
+
+    lockContractDataGen(token_id) {
+        return [{ type: Constants.TOKEN_ID_TYPE, value: token_id}]
+    }
+
+    issueDataGen(amount, unity) {
```

Comment:
change all data generator methods data structure that `processData` can process the structure directly. Remove `processFunctionData` method.